### PR TITLE
Add functionality to setup maven binary

### PR DESCRIPTION
**Pull request by bgzzz**
Add functionality to setup maven in order to use for
project building

**Description:**
Describe your changes.

**Related issue:**
Add link to the related issue.

**Check list:**
- [ ] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "@actions/http-client": "^1.0.9",
     "@actions/io": "^1.0.0",
     "@actions/tool-cache": "^1.6.1",
+    "jtest": "^0.0.1",
```
Comment (blacksails) [inline package.json:32]: what is this?

File: src/maven.ts [hunk 2]
```diff
@@ -15,11 +15,11 @@ export interface MavenOpts {
 
 export function validateOptions(opts: MavenOpts): boolean {
```
Comment (blacksails) [inline src/maven.ts:16]: `validateOptions()` and `isValidOptions()` are basically identical.

File: src/installer.ts [hunk 3]
```diff
@@ -11,6 +11,51 @@ import * as util from './util';
...
 const IS_WINDOWS = util.isWindows();
 
+export async function getMaven(version: string): Promise<void> {
+  const tmpVersions = version.split('.');
+  if (tmpVersions.length < 2) {
+    throw new Error(
+      `wrong version is set(${version}): supported version template: major.minor.patch`
+    );
+  }
+  const majorVersion = tmpVersions[0];
```
Comment (blacksails) [inline src/installer.ts:21]: You could use the 'semver' library for parsing semver.

**Review by blacksails — CHANGES_REQUESTED** [review]
Other stuff not related to lines:
- [x] I don't see a call to `core.addPath()` isen't that required in order to be able to call maven?
- [ ] You should set the gitattributes for the generated files, for an easier review

File: __tests__/installer.test.ts [hunk 4]
```diff
@@ -156,4 +156,32 @@ describe('installer tests', () => {
...
   });
+
+  it('wrong maven version of Maven has to throw error', async () => {
+    let thrown = false;
+    try {
+      await installer.getMaven('2-3-4');
+    } catch (e) {
+      thrown = true;
+    }
+    expect(thrown).toBe(true);
```
[the remark this answers is not in the document]
Comment (bgzzz) [reply __tests__/installer.test.ts:167]: this on does not work. it did not catch the error and it fails before expect evaluates. may be there is way to overcame it.    but ```expect(async () => { await installer.getMaven('2-3-4') }).toThrow()``` is not working

**Review by blacksails — APPROVED** [review]

**Pull request merged**